src/cow-react/modules/swap/containers/Row/RowFee/index.tsx

```diff
@@ -71,14 +81,16 @@ export function RowFee({ trade, fee, feeFiatValue, allowsOffchainSigning, showHe
     const feeAmountWithCurrency = `${smartFeeTokenValue} ${formatSymbol(feeCurrencySymbol)} ${
       isEthFLow ? ' + gas' : ''
     }`
-    const feeToken = smartFeeTokenValue ? feeAmountWithCurrency : '🎉 Free!'
+    console.log(smartFeeTokenValue, smartFeeFiatValue)
+    const feeToken = isValidNonZeroAmount(smartFeeTokenValue) ? feeAmountWithCurrency : '🎉 Free!'
```

Comment:
this is not part of this PR, so no need to address it, but for me is a red-flag to see a `feeToken` var that might be an amount or a label.

We are in a smart component, I believe the responsibility should be to get the value of the fee and pass it down. 
It should end up arriving to a dumb component, which will have the logic of 

if (noFee) {
  // Show is free label
} else {
   // show fee
}

Reply:
I think a small refactor could be nice here, I agree wholeheartedly with this. Perhaps with a hook that contains logic & a pure component?